packages/strapi-plugin-users-permissions/middlewares/provider/index.js

```diff
@@ -24,6 +24,21 @@ module.exports = strapi => {
 
       const grant = new Grant(strapi.plugins['users-permissions'].config.grant);
 
+      strapi.app.use(async (ctx, next) => {
```

Comment:
Adding a policy to the route `/connect` will be drastically better than use a middleware for that.